completions.go

```diff
@@ -0,0 +1,146 @@
+	PresencePenalty float32 `json:"presence_penalty"`
+	// FrequencyPenalty number between 0 and 1 that penalizes tokens on existing frequency in the text so far.
+	FrequencyPenalty float32 `json:"frequency_penalty"`
+	// BestOf generates n completions server-side and returns the "best" one.
+	BestOf *int `json:"best_of,omitempty"`
+	// LogitBias is a list of token logit biases to apply before sampling.
+	LogitBias []interface{} `json:"logit_bias,omitempty"`
+	// User is the user ID to associate with this request
+	User string `json:"user,omitempty"`
+
+	// Whether to stream back results or not. Don't set this value in the request yourself
+	// as it will be overridden depending on if you use CompletionStream or Completion methods.
+	Stream bool `json:"stream,omitempty"`
+}
+
+// CompletionResponse is the full response from a request to the completions API
+type CompletionResponse struct {
+	ID      string                     `json:"id"`
+	Object  string                     `json:"object"`
+	Created int                        `json:"created"`
+	Model   string                     `json:"model"`
+	Choices []CompletionResponseChoice `json:"choices"`
+	Usage   CompletionResponseUsage    `json:"usage"`
+}
+
+// CompletionResponseChoice is one of the choices returned in the response to the Completions API
+type CompletionResponseChoice struct {
+	Text         string        `json:"text"`
+	Index        int           `json:"index"`
+	LogProbs     LogprobResult `json:"logprobs"`
+	FinishReason string        `json:"finish_reason"`
+}
+
+// LogprobResult represents logprob result of Choice
+type LogprobResult struct {
+	Tokens        []string             `json:"tokens"`
+	TokenLogprobs []float32            `json:"token_logprobs"`
+	TopLogprobs   []map[string]float32 `json:"top_logprobs"`
+	TextOffset    []int                `json:"text_offset"`
+}
+
+// CompletionResponseUsage is the object that returns how many tokens the completion's request used
+type CompletionResponseUsage struct {
+	PromptTokens     int `json:"prompt_tokens"`
+	CompletionTokens int `json:"completion_tokens"`
+	TotalTokens      int `json:"total_tokens"`
+}
+
+// Completion is a single completion.
+func (c *client) Completion(ctx context.Context, request CompletionRequest) (*CompletionResponse, error) {
+	request.Stream = false
+	// Check if the request provides a model
+	if request.Model == "" {
+		request.Model = c.defaultModel
+	}
+	req, err := c.newRequest(ctx, "POST", "/completions", request)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := c.performRequest(req)
+	if err != nil {
+		return nil, err
+	}
+
+	output := new(CompletionResponse)
+	if err := getResponseObject(resp, output); err != nil {
+		return nil, err
+	}
+	return output, nil
+}
+
+func (c *client) CompletionStream(ctx context.Context, request CompletionRequest, onData func(*CompletionResponse)) error {
+	request.Stream = true
+	req, err := c.newRequest(ctx, "POST", "/completions", request)
```

Comment:
we probably want to apply the default model here the same as we do for `Completion` above.